TechnicoBackend/Controllers/RepairController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TechnicoBackend.Services;
+using TechnicoBackend.Models;
+using TechnicoBackend.DTOs;
+
+namespace TechnicoBackend.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class RepairController : ControllerBase
+    {
+        private readonly RepairService _repairService;
+        private readonly PropertyService _propertyService; // Προσθήκη PropertyService
+
+        // Constructor με Injection και του PropertyService
+        public RepairController(RepairService repairService, PropertyService propertyService)
+        {
+            _repairService = repairService;
+            _propertyService = propertyService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetRepairs()
+        {
+            try
+            {
+                var repairs = await _repairService.GetAllRepairsAsync();
+                return Ok(repairs);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetRepair(Guid id)
+        {
+            try
+            {
+                var repair = await _repairService.GetRepairByIdAsync(id);
+                return Ok(repair);
+            }
+            catch (KeyNotFoundException ex)
+            {
+                return NotFound(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> AddRepair([FromBody] RepairDTO repairDto)
```

Comment:
Consider using a Service to implement any business logic/functionality. Consider using Controller to only receive the data and call the service.